scripts/bash/mg_external.sh

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env bash
+
+set -x
+
+dbglog() {
+    # Allow the input to be piped
+    declare msg=${1:-$(</dev/stdin)}
+
+    echo -e "${msg}" | tee -a "${BASE_COLLECTION_PATH}"/ceph/gather-debug.log
+}
+
+
+# Expect base collection path as an exported variable
+# If it is not defined, use PWD instead
+BASE_COLLECTION_PATH=${1:-"$(pwd)"}
+echo $BASE_COLLECTION_PATH
+
+KUBECONFIG=${2:-"~/.kube/config"}
+ns=${3:-"openshift-storage"}
+
+TOOL_POD_NAME=$(oc get pods --no-headers -n ${ns} -l app='rook-ceph-tools' | awk '{print $1}')
+if [ -z "$TOOL_POD_NAME" ]; then
+    dbglog "No tool pod found"
+    echo "No tool pod found"
+    exit 2
+fi
+
+
+gather_common_ceph_resources "${BASE_COLLECTION_PATH}"
+CEPH_GATHER_DBGLOG="${BASE_COLLECTION_PATH}"/gather-ceph-debug.log
+CEPH_COLLECTION_PATH="${BASE_COLLECTION_PATH}/ceph"
+COMMAND_OUTPUT_DIR=${CEPH_COLLECTION_PATH}/must_gather_commands
+COMMAND_JSON_OUTPUT_DIR=${CEPH_COLLECTION_PATH}/must_gather_commands_json_output
+COMMAND_ERR_OUTPUT_DIR=${CEPH_COLLECTION_PATH}/logs
+mkdir -p "${COMMAND_OUTPUT_DIR}"
+mkdir -p "${COMMAND_JSON_OUTPUT_DIR}"
+mkdir -p "${COMMAND_ERR_OUTPUT_DIR}"
+
+
+pids_ceph=()
+
+# Ceph commands1
+ceph_commands1=()
+ceph_commands1+=("ceph auth list")
+ceph_commands1+=("ceph balancer pool ls")
+ceph_commands1+=("ceph balancer status")
+ceph_commands1+=("ceph config dump")
+ceph_commands1+=("ceph config-key ls")
+ceph_commands1+=("ceph crash ls")
+ceph_commands1+=("ceph crash stat")
+ceph_commands1+=("ceph device ls")
+ceph_commands1+=("ceph df detail")
+ceph_commands1+=("ceph fs dump")
+ceph_commands1+=("ceph fs ls")
+ceph_commands1+=("ceph fs status")
+ceph_commands1+=("ceph health detail")
+ceph_commands1+=("ceph healthcheck history ls")
+ceph_commands1+=("ceph mds stat")
+ceph_commands1+=("ceph mgr dump")
+ceph_commands1+=("ceph mgr module ls")
+ceph_commands1+=("ceph mgr services")
+ceph_commands1+=("ceph mon stat")
+ceph_commands1+=("ceph mon dump")
+ceph_commands1+=("ceph osd df tree")
+ceph_commands1+=("ceph osd tree")
+ceph_commands1+=("ceph osd stat")
+ceph_commands1+=("ceph osd dump")
+ceph_commands1+=("ceph osd utilization")
+ceph_commands1+=("ceph osd crush show-tunables")
+ceph_commands1+=("ceph osd crush dump")
+ceph_commands1+=("ceph osd crush weight-set ls")
+
+# Ceph commands2
```

Comment:
why do we need 2 arrays of ceph_commands instead of one

Reply:
I run all commands in parallel, and I don’t want the script to fail.

Reply:
for running all command in parallel, why do we need ceph_commands1 and ceph_commands2 ?

Reply:
I wanted to split the list because running certain `oc` commands in parallel causes the [jslave] VM to consume more resources.